src/test/java/uk/ac/ebi/subs/checklistservice/services/internal/archivechecklists/EnaChecklistServiceTest.java

```diff
@@ -0,0 +1,173 @@
+    private String localCopyDir;
+
+    @MockBean
+    private RestTemplate restTemplate;
+
+    @MockBean
+    private UsiChecklistGeneratorService usiChecklistGeneratorService;
+
+    @Autowired
+    private ChecklistRepository checklistRepository;
+
+    @Autowired
+    private EnaChecklistService enaChecklistService;
+
+    private List<String> expectedChecklistIds;
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        clearChecklistDirectory();
+    }
+
+    @Before
+    public void before() throws IllegalAccessException, IOException {
+        mockExternalResources();
+
+        checklistRepository.deleteAll();
+    }
+
+    @After
+    public void after() throws IOException {
+        clearChecklistDirectory();
+    }
+
+    @Test
+    public void testSuccessful() throws IOException {
+        Assert.assertNotNull(enaChecklistService);
+
+        List<String> updatedChecklistIds = enaChecklistService.getUpdatedChecklists();
+
+        List<Path> allFiles = Files.list(Paths.get(localCopyDir)).collect(Collectors.toList());
+
+        updatedChecklistIds.forEach(checklistId -> {
+            Assert.assertTrue(expectedChecklistIds.contains(checklistId));
+
+            Assert.assertNotNull(checklistRepository.findOne(checklistId));
+
+            Assert.assertTrue(allFiles.stream()
+                    .anyMatch(filePath -> filePath.toFile().getName().contains(checklistId)));
+        });
+    }
+
+    @Test
+    public void testUpdateAfterFailure() throws IOException {
+        Assert.assertNotNull(enaChecklistService);
+
+        // choose a checklist id.
+        String checklistId = expectedChecklistIds.get(0);
+        Assert.assertNotNull(checklistId);
+
+        // reconfigure mocking
+        String beforeRemock = usiChecklistGeneratorService.generate(checklistId);
+        Mockito.when(usiChecklistGeneratorService.generate(checklistId))
+                .thenThrow(new RuntimeException("Something went wrong. Eh."))
+                .thenReturn(beforeRemock);
+
+        // make sure the err'ed checklist id is not present afterwards.
+        Assert.assertFalse(enaChecklistService.getUpdatedChecklists().contains(checklistId));
+        Assert.assertNull(checklistRepository.findOne(checklistId));
+
+        // assert that file did get created even when the generation failed.
+        Assert.assertTrue(Files.list(Paths.get(localCopyDir))
+                .anyMatch(filePath -> filePath.toFile().getName().contains(checklistId)));
+
+        // assert that checklist updated when generation succeeded on second attempt.
+        Assert.assertTrue(enaChecklistService.getUpdatedChecklists().contains(checklistId));
+        Assert.assertNotNull(checklistRepository.findOne(checklistId));
+    }
+
+    private void mockExternalResources() throws IOException, IllegalAccessException {
+        Field summaryUrlField = ReflectionUtils.findField(EnaChecklistService.class, "CHECKLIST_SUMMARY_URL");

```

Comment:
Yes, good point. I was originally a little confused about whether I should keep these urls in the properties file or as static final class field. I decided to go with the latter because they are used after a little string manipulation and I did nt want to spend anytime on fixing the urls in case they get configured incorrectly in the config file.